tests/runtest.py

```diff
@@ -44,11 +90,283 @@
+            if config["name"] == self.unique_name:
+                configurations[index] = configuration
+                config_exists = True
+
+        if not config_exists:
+            configurations.append(configuration)
+        json_str = json.dumps(launch_json,
+                              indent=4, 
+                              separators=(',', ': '))
+
+        with open(launch_json_location, 'w') as file_handle:
+            file_handle.write(json_str)
+
+    def __create_repro_wrapper__(self):
+        """ Create the repro wrapper
+        """
+
+        if self.host_os == "Windows_NT":
+            self.__create_batch_wrapper__()
+        else:
+            self.__create_bash_wrapper__()
+
+    def __create_batch_wrapper__(self):
+        """ Create a windows batch wrapper
+        """
+    
+        wrapper = \
+"""@echo off
+REM ============================================================================
+REM Repro environment for %s
+REM 
+REM Notes:
+REM 
+REM This wrapper is automatically generated by runtest.py. It includes the
+REM necessary environment to reproduce a failure that occured during running
+REM the tests.
+REM
+REM In order to change how this wrapper is generated, see
+REM runtest.py:__create_batch_wrapper__(). Please note that it is possible
+REM to recreate this file by running tests/runtest.py --analyze_results_only
+REM with the appropriate environment set and the correct arch and build_type
+REM passed.
+REM
+REM ============================================================================
+
+REM Set Core_Root if it has not been already set.
+if "%%CORE_ROOT%%"=="" set CORE_ROOT=%s
+
+echo Core_Root is set to: "%%CORE_ROOT%%"
+
+""" % (self.unique_name, self.core_root)
+
+        line_sep = os.linesep
+
+        if self.env is not None:
+            for key, value in self.env:
+                wrapper += "echo set %s=%s%s" % (key, value, line_sep)
+                wrapper += "set %s=%s%s" % (key, value, line_sep)
+
+        wrapper += "%s" % line_sep
+        wrapper += "echo call %s%s" % (self.test_location, line_sep) 
+        wrapper += "call %s%s" % (self.test_location, line_sep) 
+
+        self.wrapper = wrapper
+    
+    def __create_bash_wrapper__(self):
+        """ Create a unix bash wrapper
+        """
+    
+        wrapper = \
+"""
+#============================================================================
+# Repro environment for %s
+# 
+# Notes:
+#
+# This wrapper is automatically generated by runtest.py. It includes the
+# necessary environment to reproduce a failure that occured during running
+# the tests.
+#
+# In order to change how this wrapper is generated, see
+# runtest.py:__create_batch_wrapper__(). Please note that it is possible
+# to recreate this file by running tests/runtest.py --analyze_results_only
+# with the appropriate environment set and the correct arch and build_type
+# passed.
+#
+# ============================================================================
```

Comment:
Could you maybe avoid duplicating all of this text? Also, this one should say __create_bash_wrapper__

Reply:
Did not want to spend too much time on this, and could not think of a super simple way to avoid this. Worth living with it for now.